src/jit/gentree.cpp

```diff
@@ -1301,6 +1301,7 @@ bool GenTree::Compare(GenTree* op1, GenTree* op2, bool swapOK)
 
                 // For the ones below no extra argument matters for comparison.
                 case GT_BOX:
+                case GT_RUNTIMELOOKUP:
```

Comment:
Had to add this because in shared code we can end up with a tree that compares a runtime lookup to a runtime lookup (for `typeof(T) == typeof(U)`). This is not hit for normal ready to run because in `Compiler::gtFoldExprCompare` there is an early out that prevents calling into `GenTree::Compare` (`(tree->gtFlags & GTF_SIDE_EFFECT)`) that is true for ready to run, but not true in CoreRT.